### ISSUE TITLE
arcade bodies do not scale with camera

### ISSUE THREAD

**Issue by fyyyyy** [photonstorm/phaser-ce#315]
This Issue is about (pick one, ✏️ delete others)

* A bug in the API:
  * Phaser version(s): 2.6.1 and earlier up to 2.8.1 and potentially higher
  * What should happen: As sprites do scale with camera, arcade bodies should scale accordingly
  * What happens instead: Sprites scale with camera, however arcade bodies stay unaffected by camera scale, leading to incorrect collisions.
* An error in the documentation: The documentation should at least hint at this problem

See also a potential fix.
https://github.com/RolandMQuiros/phaser/commit/f732cc1129c28f1cd821e28259f942f34a11c1a6

**Comment by samme** [comment]
Yes, it's always worked this way. <https://codepen.io/samme/pen/OWyPqq>

Maybe we can scale the body by `worldScale` instead.

**Comment by fyyyyy** [comment]
Is it possible to attache listeners to the camera ?
So that entities can subscribe to a camera onScale event and scale their bodies accordingly.
This would make it easier to update all bodies in the world.

**Comment by samme** [comment]
That looks good. You can also pass params like

```javascript
this.game.camera.scale.onChange.dispatch(this.game.camera.scale);
// …
updateBody(cameraScale) {
  this.body.setSize(/* … */);
}
```

**Comment by fyyyyy** [comment]
ok thanks, here is an example in case anyone is curious.
https://codepen.io/anon/pen/rzJRjK?editors=0010

**Comment by samme** [comment]
I made a branch for this, substituting `sprite.worldScale` for `sprite.scale`. The body size was correct but the offset was not, so I guess I've missed something.

**Comment by fyyyyy** [comment]
I guess all my sprites had anchor 0.5, thats why there wasnt any offset issues when scaling

**Comment by samme** [comment]
I think it's complicated because with `worldScale` the scale origin is the the top-left of the group (usually [0, 0]), whereas with `scale` it's the sprite's anchor.

**Comment by fyyyyy** [comment]
maybe it should work like the `fixedToCamera: true` property as in `scaleWithCamera: true` or `scaleWithWorld: true`
Then users can set it individually for groups or sprites, by default its false to not break existing code.

**Comment by samme** [comment]
I think the formula in https://github.com/photonstorm/phaser-ce/blob/v2.8.5/src/physics/arcade/Body.js#L618 needs to be changed somehow if `worldScale` is substituted there.

**Comment by hexus** [comment]
The real fix here is not to scale physics at all, but instead detach camera scaling from physics entirely.

https://github.com/hexus/phaser-ce/compare/master...hexus:camera-transform#diff-b0c3a19f2ef556b8df88c8078ae81441R135

This is a branch I've been working on, but I've only been hesitant to make a PR because it's a big change and will need some testing across all examples.

**Comment by samme** [comment]
Well there's always <https://codepen.io/samme/full/ZLbGzB/> 😸

**Comment by hexus** [comment]
Looks great, but changes the size of the canvas! Ouch!

**Comment by samme** [comment]
<https://codepen.io/samme/pen/vegxZz>

**Comment by hexus** [comment]
Also looks pretty nice, though it's using CSS and the world crops if you zoom out further than 1x.

I still think the best solution is to decouple the camera from the physics, as I've implemented. :)

**Comment by samme** [comment]
`body.syncBounds` may help with this.

**Comment by samme** [comment]
[Camera image from render texture](https://codepen.io/samme/pen/pvoQZrP?editors=0110) addresses this problem.
